sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala

```diff
@@ -162,7 +160,7 @@ class DataSourceV2Strategy(session: SparkSession) extends Strategy with Predicat
       }
 
     case RefreshTable(r: ResolvedTable) =>
-      RefreshTableExec(r.catalog, r.identifier, invalidateCache(r, recacheTable = true)) :: Nil
```

Comment:
@sunchao Would be better if `REFRESH TABLE` refreshes everything that depends on the table instead of dropping from the cache, isn't it?

Reply:
Yes we considered this previously (see comments [here](https://github.com/apache/spark/pull/30187#discussion_r515237470)), but it's a new behavior change so we didn't do it. I also opened [SPARK-33311](https://issues.apache.org/jira/browse/SPARK-33311) to track related efforts.

Reply:
Also, I think we should apply the approach proposed in this PR to `CatalogImpl.refreshTable()`:
https://github.com/apache/spark/blob/62d82b5b270c2eea27ba026e66b7d6598bbaa0a6/sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala#L538-L558

Current implementation is expensive in terms of Hive catalog calls and files listen. For example, when I started calling the function, it lead to additional ops in https://github.com/apache/spark/pull/31066#discussion_r556497006

Reply:
I do believe we can do `recacheByCondition` and clear data for the cached table, and everything that depend on the table in the cache. Probably, we can avoid calls to Hive metastore and rediscovering files. Though, need to experiment with that.